torch_encoder.py

```diff
@@ -66,22 +56,20 @@ def __init__(
         self.default_batch_size = default_batch_size
         self.use_default_preprocessing = use_default_preprocessing
 
-        self.default_traversal_paths = default_traversal_paths or self.DEFAULT_TRAVERSAL_PATHS
+        self.default_traversal_path = default_traversal_path
 
         # axis 0 is the batch
         self._default_channel_axis = 1
         self.model_name = model_name
-        if pool_strategy not in ('mean', 'max'):
-            raise NotImplementedError(f'unknown pool_strategy: {self.pool_strategy}')
-        self.pool_strategy = pool_strategy
-        self.pool_fn = getattr(np, self.pool_strategy)
 
         model = getattr(models, self.model_name)(pretrained=True)
 
         self.model = self._extract_feature_from_torch_module(model)
         self.model.to(torch.device(self.device))
-        if self.pool_strategy is not None:
-            self.pool_fn = getattr(np, self.pool_strategy)
+
+        self._pooling_layer = nn.AdaptiveAvgPool2d(1)
```

Comment:
```suggestion
        self._pooling_layer = nn.AdaptiveAvgPool2d(output_size=(1, 1))
```
This should be without `squeeze`